File: LeetCode-Cpp/LongestPalindromicSubstring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if (s.empty()) return "";

        int start = 0, maxLen = 1;
        int n = s.size();

        auto expand = [&](int left, int right) {
            while (left >= 0 && right < n && s[left] == s[right]) {
                left--;
                right++;
            }
            // Döngüden çıkınca bir adım fazla gitmiş oluruz, bu yüzden 1 geri alıyoruz
            int len = right - left - 1;
            if (len > maxLen) {
                maxLen = len;
                start = left + 1;
            }
        };

        for (int i = 0; i < n; i++) {
            // Tek merkezli palindrome (aba)
            expand(i, i);
            // Çift merkezli palindrome (abba)
            expand(i, i + 1);
        }

        return s.substr(start, maxLen);
    }
};

File: LeetCode-Python/LongestPalindromicSubstring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if not s:
            return ""

        start, max_len = 0, 1
        n = len(s)

        def expand(left: int, right: int):
            nonlocal start, max_len
            while left >= 0 and right < n and s[left] == s[right]:
                left -= 1
                right += 1
            length = right - left - 1
            if length > max_len:
                max_len = length
                start = left + 1

        for i in range(n):
            # Tek merkezli palindrome (örneğin "aba")
            expand(i, i)
            # Çift merkezli palindrome (örneğin "abba")
            expand(i, i + 1)

        return s[start:start + max_len]
